src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -393,16 +402,32 @@ def with_subqueries(self, subquery_plans: List["SnowflakePlan"]) -> "SnowflakePl
             df_aliased_col_name_to_real_col_name=self.df_aliased_col_name_to_real_col_name,
         )
 
-    @cached_property
+    @property
+    def quoted_identifiers(self) -> List[str]:
+        # If _attributes is not None, retrieve quoted_identifiers from it.
+        # If _attributes is None, retrieve quoted_identifiers from _quoted_identifiers.
+        # If _quoted_identifiers is None, retrieve quoted_identifiers from attributes
+        # (which triggers describe query).
+        if self._attributes is not None:
```

Comment:
what we can do here is first check self._quoted_identifiers is not None, else just return [attr.name for attr in self.attributes]. self.attributes should take care of the attribute check

Reply:
yes good idea, we can use it in next PR